test/httpClient.test.js

```diff
@@ -400,6 +400,56 @@ test('client customiseRequest function overwrites the headers and body', (t) =>
   client.destroy()
 })
 
+test('http client should wait responses', (t) => {
+  t.plan(1)
+
+  let cnt = 0
+  const opts = server.address()
+
+  opts.waitAllResponses = true
+  opts.rate = 1
+
+  const client = new Client(opts)
+
+  client.on('response', () => {
+    cnt++
+  })
+
+  client.on('done', () => {
+    t.equal(cnt, opts.rate + 1, 'response count match')
+    t.end()
+  })
+
+  setTimeout(() => {
+    client.destroy()
+  }, 1000)
```

Comment:
Maybe we can control this from the server instead? `setTimeout` is very brittle in tests.

Reply:
How do you propose to do this? We need to test waiting for a response from the server when there are input parameters: `duration` and `rate`. In this case, `setTimeout` acts as a `duration`. Or it is necessary to describe tests in `run.test.js`